### DISCUSSION TITLE
MoodNFT mints successfully but the output logs is empty

### DISCUSSION THREAD

**Discussion by smith-mern** [Cyfrin/foundry-full-course-cu#848] [General]
Cast send command runs but there is no log i am running local anvil  

`blockHash               0xc35e85759845b0e89906bc49bc1904de29fe1c36a93fc7518ed0b5085f7c7a1c
blockNumber             1402
contractAddress         
cumulativeGasUsed       21064
effectiveGasPrice       3000000007
gasUsed                 21064
logs               []
logsBloom               0x00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000
root                    
status                  1
transactionHash         0xad43657ed47a9a8ca88c72f4cc3a6aff9e9012a017e8c8b44b065980e0d8b113
transactionIndex        0
type                    2`

I'm having trouble importing an NFT into MetaMask, and I'm encountering an error message that says: "NFT can’t be added as the ownership details do not match. Make sure you have entered correct information."

I've followed the tutorial and implement the code. The contract code can be found in my GitHub repository: https://github.com/smith-mern/foundry-Nft.

I suspect that the issue might be related to how I'm calling the mintNft() function. 

`function mintNft() public {
        _safeMint(msg.sender, s_tokenCounter);
        s_tokenIdToMood[s_tokenCounter] = Mood.HAPPY;
        s_tokenCounter++;
    }`

i have the same issue when deploying to the sepolia testnet 

kindly advice

**Comment by theirrationalone** [comment]
Please confirm at least once that you're deploying MoodNft contract using a deploy script scoped inside `vm.startBroadcast(pk)` with...
```bash
forge script fileDir:contractName --rpc-url $RPC_URL --private-key $PRIVATE_KEY --verify --etherscan-api-key $ETHERSCAN_API_KEY --broadcast
```

**Reply by smith-mern** [reply]
Yes i am deploying the moodnft 


`[⠢] Compiling...
No files changed, compilation skipped
Script ran successfully.

== Return ==
0: contract MoodNft 0x9fE46736679d2D9a65F0992F2272dE9f3c7fa6e0

== Logs ==
  happySVG <svg viewBox="0 0 200 200" width="400" height="400" xmlns="http://www.w3.org/2000/svg">
    <circle cx="100" cy="100" fill="yellow" r="78" stroke="black" stroke-width="3" />
    <g class="eyes">
        <circle cx="61" cy="82" r="12" />
        <circle cx="127" cy="82" r="12" />
    </g>
    <path d="m136.81 116.53c.69 26.17-64.11 42-81.52-.73" style="fill:none; stroke: black; stroke-width: 3;" />
</svg>
  sadSVG <?xml version="1.0" standalone="no"?>
<svg width="1024px" height="1024px" viewBox="0 0 1024 1024" xmlns="http://www.w3.org/2000/svg">
    <path fill="#333"
        d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z" />
    <path fill="#E6E6E6"
        d="M512 140c-205.4 0-372 166.6-372 372s166.6 372 372 372 372-166.6 372-372-166.6-372-372-372zM288 421a48.01 48.01 0 0 1 96 0 48.01 48.01 0 0 1-96 0zm376 272h-48.1c-4.2 0-7.8-3.2-8.1-7.4C604 636.1 562.5 597 512 597s-92.1 39.1-95.8 88.6c-.3 4.2-3.9 7.4-8.1 7.4H360a8 8 0 0 1-8-8.4c4.4-84.3 74.5-151.6 160-151.6s155.6 67.3 160 151.6a8 8 0 0 1-8 8.4zm24-224a48.01 48.01 0 0 1 0-96 48.01 48.01 0 0 1 0 96z" />
    <path fill="#333"
        d="M288 421a48 48 0 1 0 96 0 48 48 0 1 0-96 0zm224 112c-85.5 0-155.6 67.3-160 151.6a8 8 0 0 0 8 8.4h48.1c4.2 0 7.8-3.2 8.1-7.4 3.7-49.5 45.3-88.6 95.8-88.6s92 39.1 95.8 88.6c.3 4.2 3.9 7.4 8.1 7.4H664a8 8 0 0 0 8-8.4C667.6 600.3 597.5 533 512 533zm128-112a48 48 0 1 0 96 0 48 48 0 1 0-96 0z" />
</svg>

EIP-3855 is not supported in one or more of the RPCs used.
Unsupported Chain IDs: 31337.
Contracts deployed with a Solidity version equal or higher than 0.8.20 might not work properly.
For more information, please see https://eips.ethereum.org/EIPS/eip-3855

## Setting up (1) EVMs.

==========================

Chain 31337

Estimated gas price: 4.7696482 gwei

Estimated total gas used for script: 3787023

Estimated amount required: 0.0180627674353086 ETH

==========================

###
Finding wallets for all the necessary addresses...
##
Sending transactions [0 - 0].
⠁ [00:00:00] [#######################################################################################################################################] 1/1 txes (0.0s)
Transactions saved to: /Users/eric/Documents/Development/foundry-f23/foundry-nft/broadcast/DeployMoodNft.s.sol/31337/run-latest.json

Sensitive values saved to: /Users/eric/Documents/Development/foundry-f23/foundry-nft/cache/DeployMoodNft.s.sol/31337/run-latest.json

##
Waiting for receipts.
⠉ [00:00:00] [###################################################################################################################################] 1/1 receipts (0.0s)
##### anvil-hardhat
✅  [Success]Hash: 0x56bcdc34d764eeffc85a4a47cd779e0c2b4801d1ac2f65bd14620482c14de9b0
Contract Address: 0x9fE46736679d2D9a65F0992F2272dE9f3c7fa6e0
Block: 3
Paid: 0.010997215583891004 ETH (2913651 gas * 3.774376404 gwei)


Transactions saved to: /Users/eric/Documents/Development/foundry-f23/foundry-nft/broadcast/DeployMoodNft.s.sol/31337/run-latest.json

Sensitive values saved to: /Users/eric/Documents/Development/foundry-f23/foundry-nft/cache/DeployMoodNft.s.sol/31337/run-latest.json



==========================

ONCHAIN EXECUTION COMPLETE & SUCCESSFUL.
Total Paid: 0.010997215583891004 ETH (2913651 gas * avg 3.774376404 gwei)

Transactions saved to: /Users/eric/Documents/Development/foundry-f23/foundry-nft/broadcast/DeployMoodNft.s.sol/31337/run-latest.json

Sensitive values saved to: /Users/eric/Documents/Development/foundry-f23/foundry-nft/cache/DeployMoodNft.s.sol/31337/run-latest.json`

**Reply by theirrationalone** [reply]
Everything seems Okay! 

However, Confirm if you're minting your nfts using some scripts Or using cmds below. 

```bash
cast send 0x9fE46736679d2D9a65F0992F2272dE9f3c7fa6e0 "mintNft()" --rpc-url http://127.0.0.1:8545 --private-key $ANVIL_PRIVATEKEY
```

**Reply by smith-mern** [reply]
i can see the logs now 
`blockHash               0x24993892af359a2c4b0df16c95f858c2fd7f79b2e16c02608ef3f76704987990
blockNumber             4
contractAddress         
cumulativeGasUsed       93037
effectiveGasPrice       3696381542
gasUsed                 93037
logs                    [{"address":"0x9fe46736679d2d9a65f0992f2272de9f3c7fa6e0","topics":["0xddf252ad1be2c89b69c2b068fc378daa952ba7f163c4a11628f55a4df523b3ef","0x0000000000000000000000000000000000000000000000000000000000000000","0x000000000000000000000000f39fd6e51aad88f6f4ce6ab8827279cfffb92266","0x0000000000000000000000000000000000000000000000000000000000000000"],"data":"0x","blockHash":"0x24993892af359a2c4b0df16c95f858c2fd7f79b2e16c02608ef3f76704987990","blockNumber":"0x4","transactionHash":"0xbaa9da8e13cff8e837cc6b8323cb67114e22cc9c217353fa3391e2210116ef26","transactionIndex":"0x0","logIndex":"0x0","transactionLogIndex":"0x0","removed":false}]
logsBloom               0x00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000040200000000000000000000000000000000000000008000000000000000000000000000000000000000000000000020000000000000100000800000000000000000000000010000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000002000000200000000000000000000000002000000000000000000020000000000000000000000000000000000000000000000000400000000000000000
root                    
status                  1
transactionHash         0xbaa9da8e13cff8e837cc6b8323cb67114e22cc9c217353fa3391e2210116ef26
transactionIndex        0
type                    2`
thanks

**Reply by theirrationalone** [reply]
You're Welcome @smith-mern !

Woww, Two MEARNS Contra here 😅. HUGEST SORRRRYYYY If that's Your Last Name and somehow i hurt you, I was taking `mern` as a Tech Word.

Pleasure to Work With You.

;`)

**Reply by smith-mern** [reply]
lol it is actually a tech word
